app/src/main/java/com/sdp13epfl2021/projmag/model/ImmutableProject.kt

```diff
@@ -90,24 +112,28 @@ data class ImmutableProject constructor(
             try {
                 val result = build(
                     id = projectId,
-                    name = map["name"] as String,
-                    lab = map["lab"] as String,
-                    teacher = map["teacher"] as String,
-                    TA = map["TA"] as String,
-                    nbParticipant = (map["nbParticipant"] as Number).toInt(),
-                    assigned = map["assigned"] as List<String>,
-                    masterProject = map["masterProject"] as Boolean,
-                    bachelorProject = map["bachelorProject"] as Boolean,
-                    tags = map["tags"] as List<String>,
-                    isTaken = map["isTaken"] as Boolean,
-                    description = map["description"] as String,
-                    videoURI = map["videoURI"] as List<String>
+                    name = map[FieldNames.NAME] as String,
+                    lab = map[FieldNames.LAB] as String,
+                    teacher = map[FieldNames.TEACHER] as String,
+                    TA = map[FieldNames.TA] as String,
+                    nbParticipant = (map[FieldNames.NB_PARTICIPANT] as Number).toInt(),
+                    assigned = map[FieldNames.ASSIGNED] as List<String>,
+                    masterProject = map[FieldNames.MASTER_PROJECT] as Boolean,
+                    bachelorProject = map[FieldNames.BACHELOR_PROJECT] as Boolean,
+                    tags = map[FieldNames.TAGS] as List<String>,
+                    isTaken = map[FieldNames.IS_TAKEN] as Boolean,
+                    description = map[FieldNames.DESCRIPTION] as String,
+                    videoURI = map[FieldNames.VIDEO_URI] as List<String>
                 )
                 return when (result) {
                     is Success -> result.value
                     is Failure -> null
                 }
-            } catch (e: Exception) {
+                /* These two exceptions occur only with corrupted Projects so should be ignored */
+            } catch (e: NullPointerException) {
+                e.printStackTrace()
+                return null
+            } catch (e: ClassCastException) {
```

Comment:
Changed to a more precise exception handling, avoiding catching the serious ones that should be reported.